### PR TITLE
Use standard modules in dyno testing

### PR DESCRIPTION
**Pull request by benharsh**
This PR modifies the compiler to remove special cases for "missing" bundled types. This PR also updates testing to use the standard library when these types are needed.

Testing:
- [x] test-dyno
- [x] local paratest

### CODE REVIEW

File: frontend/test/resolution/testCatch.cpp [hunk 1]
```diff
@@ -376,7 +375,6 @@ static void test11(Parser* parser) {
                           }
                         }
               )"""";
-  ctx->advanceToNextRevision(false);
```
Comment (DanilaFe) [inline frontend/test/resolution/testCatch.cpp:379]: Can we thread through and re-use a Context to be able to use `advanceToNextRevision` and thus save ourselves some work?

**Review by DanilaFe — COMMENTED** [review]
I see a lot of invocations of `buildStdContext()`, and each of those, if types is enabled, is going to require at least some resolution of standard modules. Can we avoid that by using `advanceToNextRevision`? I see that you've actually taken out calls to `advanceToNextRevision`, presumably because they broke things; however, I think that what's missing is that bundled search paths need to be re-set every time you advanced revisions, since they are just another query. Thus, while I would expect `advanceToNextRevision` to break standard contexts, `ADVANCE_PRESERVING_STANDARD_MODULES_` , defined in `frontend/test/test-resolution.h`, might work.

I'd also be interested to know how this affected the amount of time it takes to run the standard tests.

File: frontend/test/resolution/testCatch.cpp (hunk 1 above)
Comment (benharsh) [reply frontend/test/resolution/testCatch.cpp:379]: ``buildStdContext`` is [itself updated](https://github.com/chapel-lang/chapel/pull/26269/files#diff-6843a0340d0ef47d59be4de1a57ed5e37804f307dd540c844ef93a6759aa56de) to invoke ``advanceToNextRevision`` and reload the standard modules.

Comment (benharsh) [reply frontend/test/resolution/testCatch.cpp:379]: In my experience thus far, it should only add a couple of seconds to a parallel ``test-frontend``. Serially it went from ~65s to ~100s on my local machine.

Comment (DanilaFe) [reply frontend/test/resolution/testCatch.cpp:379]: Oh, beautiful. I missed that.

**Review by DanilaFe — APPROVED** [review]
Looks good, thank you!

**Pull request merged**